decoder/reference_targets.go

```diff
@@ -383,13 +392,18 @@ func bodySchemaAsAttrTypes(bodySchema *schema.BodySchema) map[string]cty.Type {
 }
 
 func (d *PathDecoder) collectInferredReferenceTargetsForBody(addr lang.Address, bAddrSchema *schema.BlockAddrSchema, body hcl.Body, bodySchema *schema.BodySchema, selfRefBodyRangePtr *hcl.Range, selfRefAddr lang.Address) reference.Targets {
-	refs := make(reference.Targets, 0)
-
-	content := ast.DecodeBody(body, bodySchema)
-	// We don't get body range for JSON here
-	// TODO? calculate or implement upstream
-	if bAddrSchema.DependentBodySelfRef && content.RangePtr != nil && selfRefBodyRangePtr == nil {
-		selfRefBodyRangePtr = content.RangePtr
+	var (
+		refs         = make(reference.Targets, 0)
+		hasLocalAddr = false
+		content      = ast.DecodeBody(body, bodySchema)
+	)
+	if bAddrSchema.DependentBodySelfRef || bAddrSchema.BodySelfRef {
+		if selfRefBodyRangePtr == nil {
+			// We don't get body range for JSON here
+			// TODO? calculate or implement upstream
+			selfRefBodyRangePtr = content.RangePtr
+		}
+		hasLocalAddr = selfRefBodyRangePtr != nil
```

Comment:
```suggestion
		collectLocalAddr = selfRefBodyRangePtr != nil
```
Nitpick - I think this name better reflects the intentions.